### PR TITLE
CI: add example to sim

### PR DESCRIPTION
**Pull request by ttnie**
Signed-off-by: nietingting <nietingting@xiaomi.com>

## Summary
add example include usrsocktest to sim
## Impact
no
## Testing
test on local

### CODE REVIEW

File: tools/ci/testrun/script/test_example/test_example.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    assert ret == 0
+
+
+def test_helloxx(p):
+    if p.board in do_not_support:
+        pytest.skip("unsupported at {}".format(p.board))
+    if p.board == "best1600_ep":
+        if p.core == "tee":
+            pytest.skip("unsupported at {}-{}".format(p.board, p.core))
+    ret = p.sendCommand("helloxx", "Hello, World!!")
```
Comment (acassis) [inline tools/ci/testrun/script/test_example/test_example.py:20]: I think the "helloxx" command doesn't return "Hello, World!!"! Did you test it?
It should return many different messages depending on which C++ features is enabled.
And finally it should return:
```
HelloWorld: mSecret=42
```

**Review by acassis — CHANGES_REQUESTED** [review]

File: tools/ci/testrun/script/test_example/test_example.py (hunk 1 above)
Comment (ttnie) [reply tools/ci/testrun/script/test_example/test_example.py:20]: I got the result like below, looks like different with yours. Image build by citest deconfig. @acassis 

nsh> helloxx
helloxx_main: Saying hello from the dynamically constructed instance
CHelloWorld::HelloWorld: Hello, World!!
helloxx_main: Saying hello from the instance constructed on the stack
CHelloWorld::HelloWorld: Hello, World!!
helloxx_main: Saying hello from the statically constructed instance
CHelloWorld::HelloWorld: Hello, World!!

File: tools/ci/testrun/script/test_example/test_example.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python3
+# encoding: utf-8
+import pytest
+
+pytestmark = [pytest.mark.common, pytest.mark.qemu]
+do_not_support = ["sabre-6quad", "rv-virt", "rv-virt64"]
```
Comment (xiaoxiang781216) [inline tools/ci/testrun/script/test_example/test_example.py:6]: why not test c++

File: tools/ci/testrun/script/test_example/test_example.py (hunk 1 above)
Comment (xiaoxiang781216) [reply tools/ci/testrun/script/test_example/test_example.py:17]: @acassis both will print to console:
https://github.com/apache/incubator-nuttx-apps/blob/master/examples/helloxx/helloxx_main.cxx#L67-L78
so it's fine to check "Hello, World!!".

File: tools/ci/testrun/script/test_example/test_example.py (hunk 2 above)
Comment (ttnie) [reply tools/ci/testrun/script/test_example/test_example.py]: > 

test on local c++ is works well on qemu, so enable on ci

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**